nucliadb/nucliadb/search/api/v1/chat.py

```diff
@@ -59,6 +65,47 @@
     },
 }
 
+MAX_TOKENS = 4000  # less than real because of prompt text size
+
+
+async def format_prompt_text(kbid: str, results: KnowledgeboxFindResults):
+    ordered_paras = []
+    for result in results.resources.values():
+        for field_path, field in result.fields.items():
+            for paragraph in field.paragraphs.values():
+                ordered_paras.append((field_path, paragraph))
+
+    ordered_paras.sort(key=lambda x: x[1].score, reverse=True)
```

Comment:
I think you should sort by the `order` attribute instead (and set `reverse=False`), since otherwise bm25 scores may be stepping over some good vector index results.